storage/rocksdb/ha_rocksdb.cc

```diff
@@ -1325,6 +1311,27 @@ static void rocksdb_set_io_write_timeout(
   RDB_MUTEX_UNLOCK_CHECK(rdb_sysvars_mutex);
 }
 
+int rocksdb_remove_checkpoint(const char *checkpoint_dir_raw) {
+  const auto checkpoint_dir = rdb_normalize_dir(checkpoint_dir_raw);
+  LogPluginErrMsg(INFORMATION_LEVEL, ER_LOG_PRINTF_MSG,
+                  "deleting temporary checkpoint in directory : %s\n",
+                  checkpoint_dir.c_str());
+
+  std::string trash_dir = std::string(rocksdb_datadir) + "/trash";
```

Comment:
IMHO this is important to deduplicate with the other trash dir occurence in `rocksdb_init_internal`, please extract a helper to get the trash dir.

```suggestion
  const auto trash_dir = std::string(rocksdb_datadir) + "/trash";
```